src/klein/resource.py

```diff
@@ -278,3 +278,97 @@ def processing_failed(failure, error_handlers):
         d.addErrback(processing_failed, self._app._error_handlers)
         d.addCallback(write_response).addErrback(log.err, _why="Unhandled Error writing response")
         return server.NOT_DONE_YET
+
+class Arguments(object):
+    """
+    A wrapper around the C{dict} type that gives easy access to the
+    C{bytes} keys set by Twisted.
+    """
+
+    def __init__(self, args=None):
+        self.arguments = {} if not args else args
+
+    def __contains__(self, item):
+        """
+        Check if the item is a key in the arguments.
+
+        @returns: C{bool}
+        """
+        item = ensure_utf8_bytes(item)
+        return item in self.arguments
+
+    def __delitem__(self, key):
+        del self.arguments[key]
+
+    def __getattr__(self, dict_attr):
+        """
+        Access attributes and functions from the underlaying dict.
+        """
+        return getattr(self.arguments, dict_attr)
+
+    def __getitem__(self, key):
+        """
+        Return the value from the arguments. Tries to be smart and gets
+        a single value if only 1 item exists in the list or entire list.
+
```

Comment:
I see you're point with ``__getitem__``, I'll change it to behave more in line with what is done currently (ie. return a ``list`` or raise ``KeyError``). If I implement``__getitem__``, then there's really no sense in the ``get`` function. I could just create a ``getOne`` and all other access can be done with ``[]``. Do you see value in allowing users to provide a ``default`` param?